pcdsdevices/sqr1.py

```diff
@@ -0,0 +1,281 @@
+"""
+Ophyd class for square-one tri-sphere motion system
+(https://www.sqr-1.com/tsp.html)
+
+This module defines Ophyd (python based hardware abstraction and
+interfacing library) classes for controlling a square-one tri-sphere motion
+system. The motion system consists of multiple axes, each of which can be
+controlled individually. The axes include X, Y, Z, rX, rY, and rZ.
+
+Classes:
+- SQR1Axis: A class representing a single axis of the tri-sphere motion system.
+It inherits from PVPositionerIsClose and includes attributes for setpoint,
+readback, actuation, and stopping the motion.
+
+- SQR1: A class representing the entire tri-sphere motion system.
+It is a Device that aggregates multiple SQR1Axis instances for each axis.
+It also includes methods for multi-axis movement and stopping the motion.
+"""
+
+from __future__ import annotations
+
+import enum
+import logging
+import typing
+
+from ophyd.device import Component as Cpt
+from ophyd.device import Device
+from ophyd.device import FormattedComponent as FCpt
+from ophyd.pv_positioner import PVPositionerIsClose
+from ophyd.signal import EpicsSignal
+from ophyd.status import wait as status_wait
+
+logger = logging.getLogger(__name__)
+
+
+class Axis(str, enum.Enum):
+    """
+    An enumeration representing axes available in tri-sphere table stage.
+    This enumeration defines the following axes:
+    - `X`: X-axis translation
+    - `Y`: Y-axis translation
+    - `Z`: Z-axis translation
+    - `rX`: X-axis rotation
+    - `rY`: Y-axis rotation
+    - `rZ`: Z-axis rotation
+    """
+
+    X = "X"
+    Y = "Y"
+    Z = "Z"
+    rX = "rX"
+    rY = "rY"
+    rZ = "rZ"
+
+
+class SQR1Axis(PVPositionerIsClose):
+    """
+    This class represents and controls a single axis of the square-one
+    tri-sphere motion system using EPICS PVs (Process Variables).
+    It is a subclass of PVPositionerIsClose and includes attributes for setting
+    the setpoint, reading back the current position,
+    actuating the motion, and stopping the motion.
+
+    Attributes:
```

Comment:
If you look at our style guide [here](https://pcdshub.github.io/style.html#docstrings), or any of the other code in this repository (or any of our other Python repositories), you'll see we follow the numpy docstring convention.

I think it would be good for you to understand how to format them for your own future reference/submissions, but we can reformat them for you if necessary to get this merged sooner rather than later.

Reply:
thanks for sharing the link, I wasn't aware of it.

Reply:
Thanks for fixing it up 👍